src/Tizen.System.Session/Session/Session.cs

```diff
@@ -0,0 +1,520 @@
+            CheckError(ret, "Interop failed to get user list");
+
+            string[] users;
+            IntPtrToStringArray(ptr, count, out users);
+            Interop.Session.SubsessionFreeUserList(ptr);
+
+            return new List<string>(users);
+        }
+
+        /// <summary>
+        /// Gets a currently active subession user ID for this session.
+        /// </summary>
+        /// <remarks>
+        /// When no subsession is enabled, "" (empty string, see EmptyUser field) is returned.
+        /// This doesn't mean that "" is a subsession ID in the same way as others; it is just a marker meaning
+        /// that no subsession is enabled.
+        /// </remarks>
+        /// <exception cref="ArgumentException">Session UID of this object is invalid</exception>
+        /// <exception cref="OutOfMemoryException">Out of memory</exception>
+        /// <exception cref="IOException">Internal error</exception>
+        /// <exception cref="UnauthorizedAccessException">Not permitted</exception>
+        /// <exception cref="NotSupportedException">Not supported</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string GetCurrentUser()
+        {
+            StringBuilder user = new StringBuilder(MaxUserLength);
+            SessionError ret = Interop.Session.SubsessionGetCurrentUser(SessionUID, user);
+            CheckError(ret, "Interop failed to get current subsession user");
+
+            return user.ToString();
+        }
+
+        /// <summary>
+        /// Request new subsession to be created.
+        /// </summary>
+        /// <param name="userName">Subesssion user ID to be created</param>
+        /// <remarks>
+        /// Subsession ID must not start with a dot or have slashes.
+        /// </remarks>
+        /// <exception cref="ArgumentException">Session UID of this object is invalid, or user ID is not a valid subession ID</exception>
+        /// <exception cref="InvalidOperationException"> Provided subsession user ID already exists</exception>
+        /// <exception cref="OutOfMemoryException">Out of memory</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Task SubsessionAddUserAsync(string userName)
+        {
+            var task = new TaskCompletionSource<bool>();
+            int taskID = 0;
+
+            lock (_replyLock)
+            {
+                taskID = _replyID++;
+            }
+
+            _replyMap[taskID] = (int result, IntPtr data) =>
+            {
+                _replyMap.Remove((int)data);
+                try
+                {
+                    CheckError((SessionError)result, "Interop failed to complete adding a new subsession user");
+                }
+                catch (Exception exception)
+                {
+                    task.SetException(exception);
+                    return;
+                }
+
+                task.SetResult(true);
+            };
```

Comment:
```suggestion
            _replyMap[taskID] = (int result, IntPtr data) =>
            {

                try
                {
                    CheckError((SessionError)result, "Interop failed to complete adding a new subsession user");
                    task.SetResult(true);
                }
                catch (Exception exception)
                {
                    task.SetException(exception);
                    return;
                }
                finally
                {
                    _replyMap.Remove((int)data);
                }
            };
```

Reply:
Thank you. Yes, using ```finally``` is definitely a good idea here.